File: Design-Patterns-Using-Swift/Bridge/BridgePattern.swift
// Abstraction protocol
protocol Abstraction {
    func operation()
}

// Implementor protocol
protocol Implementor {
    func operation()
}

// ConcreteImplementor1 class
class ConcreteImplementor1: Implementor {
    func operation() {
        print("ConcreteImplementor1 operation")
    }
}

// ConcreteImplementor2 class
class ConcreteImplementor2: Implementor {
    func operation() {
        print("ConcreteImplementor2 operation")
    }
}

// ConcreteAbstraction class
class ConcreteAbstraction: Abstraction {
    private var imp: Implementor

    init(imp: Implementor) {
        self.imp = imp
    }

    func operation() {
        imp.operation()
    }
}

// Client code
let c1 = ConcreteImplementor1()
let abstraction: Abstraction = ConcreteAbstraction(imp: c1)
abstraction.operation()

/*
ConcreteImplementor1 operation
*/

File: Design-Patterns-Using-Ruby/Bridge/BridgePattern.rb
# Abstraction interface
class Abstraction
    def initialize(imp)
        @imp = imp
    end
  
    def operation
        @imp.operation
    end
end

# Implementor interface
class Implementor
    def operation
        raise NotImplementedError, 'Subclasses must implement this method'
    end
end

# ConcreteImplementor1 class
class ConcreteImplementor1 < Implementor
    def operation
        puts 'ConcreteImplementor1 operation'
    end
end

# ConcreteImplementor2 class
class ConcreteImplementor2 < Implementor
    def operation
        puts 'ConcreteImplementor2 operation'
    end
end

# ConcreteAbstraction class
class ConcreteAbstraction < Abstraction
end

# Client code
imp = ConcreteImplementor1.new
abstraction = ConcreteAbstraction.new(imp)
abstraction.operation

=begin
ConcreteImplementor1 operation
=end